security/fma-interop-access-list.md

```diff
@@ -0,0 +1,121 @@
+
+Below are references for this project:
+
+- Design PR: [Access List design](https://github.com/ethereum-optimism/design-docs/pull/214)
+- Specs PR: [Access List Validation](https://github.com/ethereum-optimism/specs/pull/612)
+
+## Failure Modes and Recovery Paths
+
+### FM1: Gas Schedule Changes Break Validation
+
+- **Description:** The validation mechanism relies on gas introspection to determine if storage slots are warm. If the EVM's gas schedule changes, particularly around storage access costs, it could break the validation logic.
+- **Risk Assessment:** High
+  - Potential Impact: Critical. Changes to gas costs could cause all cross-chain message validation to fail or allow invalid messages to pass.
+  - Likelihood: Low. While gas schedule changes are rare, they do occur during network upgrades.
+- **Mitigations:**
+  - Set `WARM_READ_THRESHOLD` conservatively to account for potential fluctuations
+  - Add tests that verify validation works across different gas cost scenarios
+  - Document gas schedule dependencies clearly for future upgrades
+- **Detection:** Monitor for network upgrades that modify storage access gas costs. Run validation tests before accepting upgrades.
+- **Recovery Path(s):** Deploy contract upgrade with adjusted thresholds if gas schedule changes significantly.
+
+### FM2: Storage Slot Collision
+
+- **Description:** The checksum calculation for message validation uses storage slots that could potentially collide with other slots from the access list, leading to false validation of messages.
+- **Risk Assessment:** High
+  - Potential Impact: Critical. Storage collisions could allow unauthorized messages to be validated.
+  - Likelihood: Low. Storage slots are carefully calculated using Identifier and message hash parameter. The way both are encoded in the contract’s logic needs to be flawed for this case to happen.
+- **Mitigations**
+  - Implement robust checksum calculation that minimizes collision risk
+  - Add tests verifying no collisions occur with expected storage patterns
+  - Document storage layout and slot calculation methodology
+- **Detection:** Monitor for unexpected message validations that could indicate storage collisions.
+- **Recovery Path(s):** Deploy contract upgrade with revised storage slot calculation if collisions are detected and update ALL off-chain tools and SDKs that generate these hashes on the access list.
+
+### FM3: Access List Manipulation
+
+- **Description:** Malicious actors could attempt to manipulate transaction access lists to force validation of unauthorized messages.
+- **Risk Assessment:** High
+  - Potential Impact: High. Could allow unauthorized message execution.
+  - Likelihood: Medium. Access lists are user-controlled transaction parameters.
+- **Mitigations:**
+  - Ensure checksum calculation is cryptographically sound
+  - Implement proper validation of access list entries
+  - Add checks for maximum access list size
```

Comment:
The action item here is to have a paramaterizable max access list size as a policy on ingress/mempool. This is similar to the max tx size for p2p gossip being smaller than the max tx size based on using all gas for calldata in a block